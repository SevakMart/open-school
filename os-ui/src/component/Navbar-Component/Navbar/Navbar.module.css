.navContainer, .categoriesMentors, .buttonContent {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  box-sizing: border-box;
}

.categoriesMentors p, .buttonContent p {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 500;
  font-size: 1rem;
  line-height: 1.5rem;
  color: #5E617B;
  cursor: pointer;
}

.navContainer {
  padding: 0 6%;
}
.navContainer h2 {
  font-family: "Helvetica Neue";
  font-style: italic;
  font-weight: 700;
  font-size: 20px;
  line-height: 24px;
  color: #222427;
  cursor: pointer;
}
.navContainer .categoriesMentors {
  width: 15%;
}
.navContainer .buttonContent {
  width: 20%;
}/*# sourceMappingURL=Navbar.module.css.map */