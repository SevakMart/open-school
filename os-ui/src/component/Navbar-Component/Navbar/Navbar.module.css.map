{"version":3,"sources":["Navbar.module.scss","../../../styles/Mixins/FlexMixins.scss","Navbar.module.css","../../../styles/Mixins/FontMixin.scss","../../../styles/Configs/Fonts.scss","../../../styles/Configs/Colors.scss"],"names":[],"mappings":"AAMA;ECLE,aAAA;EACA,mBDKmB;ECJnB,mBDIwB;ECHxB,8BDGgC;EAChC,sBAAA;AEFF;;AFIA;EGTE,yBCDY;EDEZ,kBHSkC;EGRlC,gBHQ0C;EGP1C,eHO+C;EGN/C,mBHMqD;EACrD,cKVgB;ELWhB,eAAA;AEGF;;AFDA;EACE,aAAA;AEIF;AFHE;EGhBA,6BAAA;EACA,kBHgBsC;EGftC,gBHe8C;EGd9C,eHcmD;EGbnD,iBHawD;EACtD,cKhBgB;ELiBhB,eAAA;AESJ;AFNE;EACE,UAAA;AEQJ;AFNE;EACE,UAAA;AEQJ","file":"Navbar.module.css"}