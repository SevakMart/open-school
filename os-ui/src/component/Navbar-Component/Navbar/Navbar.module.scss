@import "../../../styles/Mixins/FlexMixins.scss";
@import "../../../styles/Mixins/FontMixin.scss";
@import "../../../styles/Mixins/ButtonMixin.scss";
@import "../../../styles/Configs/Colors.scss";
@import "../../../styles/Configs/Fonts.scss";
@import "../../../styles/Configs/PxToRemCalculator.scss";

.navContainer,.categoriesMentors,.buttonContent {
  @include flexStyle(row, center, space-between);
  box-sizing:border-box;
}
.categoriesMentors p,.buttonContent p {
  @include fontStyle($primary-font, normal, 500, 1rem, 1.5rem);
  color:$main-text-color;
  cursor:pointer;
}
.navContainer{
  padding:0 6%;
  h2 {
    @include fontStyle($secondary-font, italic, 700, 20px,24px);
    color:$header-main-color;
    cursor:pointer;
  }
  
  .categoriesMentors {
    width:15%;
  }
  .buttonContent {
    width: 20%;
  }
}
