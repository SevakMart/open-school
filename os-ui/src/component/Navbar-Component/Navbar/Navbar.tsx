import { useTranslation } from 'react-i18next';
import { useDispatch } from 'react-redux';
import { useContext } from 'react';
import Button from '../../Button/Button';
import { Types } from '../../../types/types';
import { openModal } from '../../../redux/Slices/PortalOpenStatus';
import styles from './Navbar.module.scss';
import { signInContext } from '../../../contexts/Contexts';

/* eslint-disable max-len */

const Navbar = () => {
  const { t } = useTranslation();
  const dispatch = useDispatch();
  const { signIn, setSignIn } = useContext(signInContext);
  const { navContainer, categoriesMentors, buttonContent } = styles;

  const handleSignIn = () => {
    dispatch(openModal({ buttonType: Types.Button.SIGN_IN }));
  };
  const goToExploreCategoriesPage = () => {
    dispatch(openModal({ buttonType: Types.Button.SIGN_IN }));
  };
  const goToMentorsPage = () => {
    dispatch(openModal({ buttonType: Types.Button.SIGN_IN, isRequestForMentorsPage: true }));
  };

  return (
    <nav className={navContainer}>
      <h2>{t('string.homePage.navBar.logo')}</h2>
      <div className={categoriesMentors}>
        <p onClick={goToExploreCategoriesPage}>{t('string.homePage.navBar.exploreCategories')}</p>
        <p onClick={goToMentorsPage}>{t('string.homePage.navBar.mentors')}</p>
      </div>
      <div className={buttonContent}>
        <p>{t('button.homePage.becomeMentor')}</p>
        <Button.SignInButton className={['navbarSignInButton']} onClick={handleSignIn}>
          {t('button.homePage.signIn')}
        </Button.SignInButton>
      </div>
    </nav>
  );
};
export default Navbar;
