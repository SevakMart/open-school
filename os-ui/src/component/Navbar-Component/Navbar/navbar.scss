@import url('https://fonts.googleapis.com/css2?family=Caveat:wght@400;500;600&family=Inconsolata:wght@400;500;600;700;800&family=Manrope:wght@300;400;500;600;700;800&family=Montserrat&display=swap');
@import "../../../styles/Configs/Fonts.scss";
@import "../../../styles/Configs/Colors.scss";

.homePage {
  font-family: $primary-font;
  padding-top: 10px;
  display: flex;
  align-items: center;
  &_Logo{
    margin-left: 80px;
    font-family: 'Inconsolata';
    font-weight: 400;
    font-size: 20px;
    line-height: 21px;
    color: $header-main-color;
    cursor: pointer;
  }
  &_leftMenu {
    margin-left: 85px;
    display: flex;
    flex: 1;
    gap: 35px;
    &_item {
      font-weight: 500;
      font-size: 16px;
      line-height: 24px;
      color: $main-text-color;
      // text-transform: uppercase;
      cursor: pointer;
    }
  }
  
  &_rightMenu {
    display: flex;
    align-items: center;
    gap: 30px;
    margin-right: 80px;
    &_item {
      font-weight: 500;
      font-size: 12px;
      line-height: 15px;
      color: $header-main-color;
      text-transform: uppercase;
      cursor: pointer;
    }
  }
  &_btn {
    font-weight: 500;
    font-size: 12px;
    line-height: 15px;
    border-radius: 20px;
    height: 35px;
    padding: 10px 15px;
    background-color: $black-text-color;
    color: $white-text-color;
    cursor: pointer;
    &:hover {
      background-color: #48505c;
    }
    &:active {
      background-color: #17191d;
    }
  }
}

// mobile
.mobileMenu {
  font-family: $primary-font;
  display: none;
}

.mobileList {
  position: absolute;
  top: 0;
  left: 0;
  width: 230px;
  height: 100%;
  background: linear-gradient(to bottom, rgba($email-icon-color, 1) 0%, rgba($email-icon-color, 0) 100%);
  opacity: 0.8;
  &_info {
    margin-top: 60px;
    margin-left: 30px;
    cursor: pointer;
    p {
      color: $header-main-color;
      font-weight: 500;
      &:hover {
        color: #e0e0e0;
      }
      &:active {
        color: #121314;
        font-weight: 600;
      }
    }
  }
  &.open {
    animation: slideDown 0.3s forwards;
  }
}

@keyframes slideDown {
  from {
    opacity: 0;
    transform: translateX(-100%);
  }
  to {
    transform: translateX(0);
  }
}
