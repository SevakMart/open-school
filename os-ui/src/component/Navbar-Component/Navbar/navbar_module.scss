@import url('https://fonts.googleapis.com/css2?family=Caveat:wght@400;500;600&family=Inconsolata:wght@400;500;600;700;800&family=Manrope:wght@300;400;500;600;700;800&family=Montserrat&display=swap');
@import "../../../styles/Configs/Fonts.scss";
@import "../../../styles/Configs/Colors.scss";

@media only screen and (max-width: 1045px) {
  .homePage {
    &_Logo{
      font-size: 16px;
      line-height: 15px;
    }
    &_leftMenu {
      margin-left: 45px;
      gap: 30px;
      &_item {
        font-weight: 400;
        font-size: 14px;
        line-height: 15px;
      }
    }
    &_rightMenu {
      &_item {
        font-weight: 400;
        font-size: 10px;
      }
    }
  }
}

@media only screen and (max-width: 900px) {
  .homePage {
    &_Logo{
      margin-left: 35px;
    }
  }
  .mobileMenu {
    display: flex;
    padding-top: 11px;
  }

  .desktopMenu {
    display: none;
  }

  .mobileMenuButton {
    margin-left: auto;
    margin-right: 40px;
    cursor: pointer;
  }
}  
