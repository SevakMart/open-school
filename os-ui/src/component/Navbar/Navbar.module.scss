@import "../../styles/Mixins/FlexMixins.scss";
@import "../../styles/Mixins/FontMixin.scss";
@import "../../styles/Mixins/ButtonMixin.scss";

.navContainer {
  @include flexStyle(row, center, space-between);
  padding-left: 6%;
  padding-right: 6%;
  h2 {
    @include fontStyle("Helvetica Neue", italic, 700, 28px, 24px);
    color: #222427;
  }
  .navMainContent {
    @include flexStyle(row, center, space-between);
    width: 40%;
  }
  .categoriesMentors {
    @include flexStyle(row, center, space-between);
    width: 45%;
    p {
      @include fontStyle("Montserrat", normal, 500, 16px, 24px);
      color: #5e617b;
      cursor: pointer;
    }
  }
  .buttonContent {
    @include flexStyle(row, center, space-between);
    width: 50%;
    & button:nth-of-type(1) {
      @include buttonStyle(
        none,
        #5e617b,
        8px,
        none,
        35px,
        16px,
        16px,
        10px,
        10px
      );
      @include fontStyle("Montserrat", normal, 500, 12px, 15px);
      text-transform: uppercase;
    }
    & button:nth-of-type(2) {
      @include buttonStyle(
        #5e617b,
        #ffffff,
        8px,
        none,
        35px,
        16px,
        16px,
        10px,
        10px
      );
      @include fontStyle("Montserrat", normal, 500, 12px, 15px);
      text-transform: uppercase;
    }
  }
}
