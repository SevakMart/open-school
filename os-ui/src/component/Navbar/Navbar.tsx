import { useTranslation } from 'react-i18next';
import { useDispatch } from 'react-redux';
import Button from '../Button/Button';
import { Types } from '../../types/types';
import { openModal } from '../../redux/Slices/PortalOpenStatus';
import styles from './Navbar.module.scss';

const Navbar = () => {
  const { t } = useTranslation();
  const dispatch = useDispatch();
  const { navContainer, categoriesMentors, buttonContent } = styles;

  const handleSignIn = () => {
    dispatch(openModal(Types.Button.SIGN_IN));
  };

  return (
    <nav className={navContainer}>
      <h2>{t('string.homePage.navBar.logo')}</h2>
      <div className={categoriesMentors}>
        <p>{t('string.homePage.navBar.exploreCategories')}</p>
        <p>{t('string.homePage.navBar.mentors')}</p>
      </div>
      <div className={buttonContent}>
        <p>{t('button.homePage.becomeMentor')}</p>
        <Button.SignInButton className={['navbarSignInButton']} onClick={handleSignIn}>
          {t('button.homePage.signIn')}
        </Button.SignInButton>
      </div>
    </nav>
  );
};
export default Navbar;
