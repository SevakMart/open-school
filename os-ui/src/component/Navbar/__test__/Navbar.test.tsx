import { render } from '@testing-library/react';
import Navbar from '../Navbar';

describe('Make unit tests of Navbar component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(<Navbar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
