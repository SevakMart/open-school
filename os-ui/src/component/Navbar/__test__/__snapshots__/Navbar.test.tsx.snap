// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make unit tests of Navbar component Create a snapshot test 1`] = `
<DocumentFragment>
  <nav
    class="navContainer"
  >
    <h2>
      Open-School
    </h2>
    <div
      class="navMainContent"
    >
      <div
        class="categoriesMentors"
      >
        <p>
          Explore Categories
        </p>
        <p>
          Mentors
        </p>
      </div>
      <div
        class="buttonContent"
      >
        <button
          type="button"
        >
          Become a mentor
        </button>
        <button
          type="button"
        >
          Sign In
        </button>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
